### PR TITLE
Fix blobstore errors for droplets from CF Docker apps

### PR DESCRIPTION
**Pull request by jochenehret**
* A short explanation of the proposed change:
PR fixes droplet deletion error messages for Docker apps. For details, see https://github.com/cloudfoundry/cloud_controller_ng/issues/2582

* An explanation of the use cases your change solves
See https://github.com/cloudfoundry/cloud_controller_ng/issues/2582

* Links to any other associated PRs

* [x] I have reviewed the [contributing guide](https://github.com/cloudfoundry/cloud_controller_ng/blob/main/CONTRIBUTING.md)

* [x] I have viewed, signed, and submitted the Contributor License Agreement

* [x] I have made this pull request to the `main` branch

* [x] I have run all the unit tests using `bundle exec rake`

* [x] I have run [CF Acceptance Tests](https://github.com/cloudfoundry/cloud_controller_ng#cf-acceptance-tests-cats)

### CODE REVIEW

File: spec/unit/lib/cloud_controller/bits_expiration_spec.rb [hunk 1]
```diff
@@ -41,19 +41,47 @@ module VCAP::CloudController
...
+          DropletModel.make(
+            app_guid:     app.guid,
+            created_at:   t + i,
+            droplet_hash: nil,
+            docker_receipt_image: 'repo/test-app',
+          )
+        end
+      end
+
+      it 'it does not enqueue a job to delete the blob' do
```
Comment (philippthun) [inline spec/unit/lib/cloud_controller/bits_expiration_spec.rb:65]: it '~~it~~ does not'

File: spec/unit/lib/cloud_controller/bits_expiration_spec.rb [hunk 2]
```diff
@@ -41,19 +41,47 @@ module VCAP::CloudController
...
+        expect { BitsExpiration.new.expire_droplets!(app) }.not_to change { Delayed::Job.count }.from(0)
```
Comment (philippthun) [inline spec/unit/lib/cloud_controller/bits_expiration_spec.rb:66]: `not_to change` does not require the `from(0)`

File: spec/unit/jobs/runtime/expired_blob_cleanup_spec.rb [hunk 3]
```diff
@@ -13,21 +13,14 @@ module Jobs::Runtime
 
       describe 'droplets' do
         context 'expired' do
-          let!(:expired_droplet) { DropletModel.make(state: DropletModel::EXPIRED_STATE) }
```
Comment (philippthun) [inline spec/unit/jobs/runtime/expired_blob_cleanup_spec.rb:16]: I think this file does not need to be changed (as also `expired_blob_cleanup.rb` did not change). It checks that the `exclude(droplet_hash: nil)` is working.

**Review by philippthun — CHANGES_REQUESTED** [review]

File: lib/cloud_controller/bits_expiration.rb [hunk 4]
```diff
@@ -20,7 +20,9 @@ def expire_droplets!(app)
 
       droplets_to_expire.each do |droplet|
         droplet.update(state: DropletModel::EXPIRED_STATE)
-        enqueue_droplet_delete_job(droplet.guid)
+        if droplet.droplet_hash
```
[the remark this answers is not in the document]
Comment (jochenehret) [reply lib/cloud_controller/bits_expiration.rb:23]: Ok

File: spec/unit/lib/cloud_controller/bits_expiration_spec.rb (hunk 1 above)
Comment (jochenehret) [reply spec/unit/lib/cloud_controller/bits_expiration_spec.rb:65]: Ok

File: spec/unit/lib/cloud_controller/bits_expiration_spec.rb (hunk 2 above)
Comment (jochenehret) [reply spec/unit/lib/cloud_controller/bits_expiration_spec.rb:66]: Ok

File: spec/unit/jobs/runtime/expired_blob_cleanup_spec.rb (hunk 3 above)
Comment (jochenehret) [reply spec/unit/jobs/runtime/expired_blob_cleanup_spec.rb:16]: Yes, but we want to emphasise that there are two types of droplets: For buildpack apps and for Docker apps. The `expired_blob_cleanup.rb` coding is ok. It correctly filters droplet db entries with `droplet_hash: nil`. This enhancement of the test prevents regressions if someone changes the filtering.

Comment (philippthun) [reply spec/unit/jobs/runtime/expired_blob_cleanup_spec.rb:16]: But there were already two tests for that:

1. [it 'enqueues a deletion job when droplet_hash is not nil'](https://github.com/cloudfoundry/cloud_controller_ng/blob/a63cfe80d92e8f153db89d10963f10f9173a2f69/spec/unit/jobs/runtime/expired_blob_cleanup_spec.rb#L19-L20)

2. [it 'does nothing when droplet_hash is nil'](https://github.com/cloudfoundry/cloud_controller_ng/blob/a63cfe80d92e8f153db89d10963f10f9173a2f69/spec/unit/jobs/runtime/expired_blob_cleanup_spec.rb#L26-L27)

**Review by philippthun — CHANGES_REQUESTED** [review]

File: spec/unit/jobs/runtime/expired_blob_cleanup_spec.rb (hunk 3 above)
Comment (jochenehret) [reply spec/unit/jobs/runtime/expired_blob_cleanup_spec.rb:16]: Yes, but there was no "Docker"-related test object in this spec. At first, droplets were only used for CF buildpack apps. Then later the Docker feature was added. The droplet concept was reused, but with a different attribute configuration (`droplet_hash: nil` and instead a `docker_receipt_image`). We wanted to express this fact more clearly in the spec by adding one Docker droplet test object.

Comment (philippthun) [reply spec/unit/jobs/runtime/expired_blob_cleanup_spec.rb:16]: Okay, now I understand your intention.

Then I suggest to put the "Docker" droplet into a `let!` and give it a descriptive name (maybe `expired_docker_droplet`). And to distinguish between these two types of droplets, I would also suggest to rename the `expired_droplet` (maybe to `expired_buildpack_droplet`?).

And as you focus more on the semantics (and not the technical details of `droplet_hash` being `nil` or not) the spec could be changed to `it 'enqueues a deletion job for the buildpack droplet only'`.

There is also another context below (`context 'failed'`) with a similar intention. This one should be changed in a similar way to be consistent.

**Review by philippthun — APPROVED** [review]

**Pull request merged**